src/libraries/System.Linq.AsyncEnumerable/src/System/Linq/Empty.cs

```diff
@@ -16,7 +16,11 @@ public static partial class AsyncEnumerable
         /// <returns>An empty <see cref="IAsyncEnumerable{T}"/> whose type argument is <typeparamref name="TResult"/>.</returns>
         public static IAsyncEnumerable<TResult> Empty<TResult>() => EmptyAsyncEnumerable<TResult>.Instance;
 
-        private sealed class EmptyAsyncEnumerable<TResult> : IAsyncEnumerable<TResult>, IAsyncEnumerator<TResult>
+        /// <summary>Determines whether <paramref name="source"/> is known to be an always-empty enumerable.</summary>
+        private static bool IsEmpty<TResult>(this IAsyncEnumerable<TResult> source) => source is EmptyAsyncEnumerable<TResult>;
```

Comment:
Would it make more sense to name this as `IsStaticEmpty`? The current name (`IsEmpty`) reads as if it is the logical equivalent of `!Any` when reading code elsewhere for example in the below excerpt, which at first glance looked as if there could be a bug.

```csharp
            return
                source.IsEmpty() ? Empty<KeyValuePair<TKey, TAccumulate>>() :
                Impl(source, keySelector, seedSelector, func, keyComparer, default);
```